test/minikube/minikube_rolling_upgrade_test.go

```diff
@@ -232,6 +240,62 @@ func TestKubernetesUpgradeFullDatabaseMinorVersion(t *testing.T) {
 
 		verifyAllProcessesRunning(t, namespaceName, admin0, 2)
 	})
+
```

Comment:
the code on lines 239-241 assumes that time is linearly moving forward. With KAA upgrade enabled, the TE might get killed while AwaitDatabaseUp is still running. So it was running, but then isn't again. This means that a) that function might observe 2 restarts and hence take a longer time (which should be fine since we wait for 300s) and b) once it finally returns, the DB version is already upgraded.

This could cause subtle timing hazards in this test.

Reply:
Good point!
Actually, there will be a problem if {{AwaitDatabaseUp}} returns before the TE is restarted and then {{verifyAllProcessesRunning}} may fail if the new TE doesn't go `MONITORED` in 30 sec.

I will enable the automatic upgrade in the sub-test and add await for the TE restart count to harden it.

Reply:
thats a really neat solution!